### ISSUE TITLE
Chore: Add `husky` to the repo

### ISSUE THREAD

**Issue by coolaj86** [webinstall/webi-installers#474]
As per <https://github.com/typicode/husky#install>:

1. Add `husky` (no transitive dependencies):
   ```sh
   npm install --save-dev husky
   ```
2. Add a new `prepare` to `package.json.scripts`:
   ```sh
   npm set-script prepare "husky install"
   npm run prepare
   ```
3. Create the new `pre-commit` script
   ```sh
   npx husky add .husky/pre-commit "npm run fmt"
   npx husky add .husky/pre-commit "npm run lint"
   git add .husky/pre-commit
   ```

**Comment by TarekArar99** [comment]
Hi @coolaj86 , I'm trying to raise a PR, but when I try to commit my code it says fmt script is missing in package.json, could I get help on that ?

**Comment by coolaj86** [comment]
Those commands need to be set / updated.

```sh
npm set-script fmt "npm run prettier; npm run shfmt"
npm set-script jshint "npx jshint@2.x -c ./.jshintrc */*.js"
npm set-script shellcheck "fd -t f -e sh | xargs -- shellcheck -s sh --exclude=SC2034,SC2129,SC2069,SC2154,SC2164,SC2016,SC2001 --"
npm set-script lint "npm run jshint; npm run shellcheck"
```

**Comment by ShyamMahajan** [comment]
@coolaj86 I have raised a PR for this one. please check

**Comment by coolaj86** [comment]
Closed via #515 and #659
